File: 2021/test/day_11_test.rb
# frozen_string_literal: true

require "test_helper"
require "day_11"

class Day11Test < Minitest::Test
  def test_part1
    day = Day11.new

    assert_equal 1656, day.part1(input("day_11_example_1"))
    assert_equal 1705, day.part1(input("day_11"))
  end

  def test_part2
    day = Day11.new

    assert_equal 195, day.part2(input("day_11_example_1"))
    assert_equal 265, day.part2(input("day_11"))
  end
end

File: 2023-elixir/test/day_11_test.exs
defmodule Day11Test do
  use ExUnit.Case

  import TestHelper
  import Day11

  test "part1" do
    assert part_1(input("day_11_example_1")) == 374
    assert part_1(input("day_11")) == 9_214_785
  end

  test "part2" do
    assert part_2(input("day_11")) == 613_686_987_427
  end
end
